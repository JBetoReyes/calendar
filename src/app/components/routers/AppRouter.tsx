/* eslint-disable import/no-named-as-default */
import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import CalendarScreen from '../calendar/CalendarScreen';
import LoginScreen from '../auth/LoginScreen';

const AppRouter = (): JSX.Element => {
  return (
    <Router>
      <div className="router-wrapper">
        <Switch>
          <Route exact path="/login" component={LoginScreen} />
          <Route exact path="/" component={CalendarScreen} />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
