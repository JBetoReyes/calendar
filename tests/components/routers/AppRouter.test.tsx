import {mount} from 'enzyme';
import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import AppRouter from '../../../src/app/components/routers/AppRouter';

const middlewares = [thunk];
const mockStoreProvider = configureStore(middlewares);
let store;
describe('AppRouter', () => {
  test('should match with snapshot', () => {
    store = mockStoreProvider({
      auth: {
        checking: true,
        uid: null,
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('should render the login screen', () => {
    store = mockStoreProvider({
      auth: {
        checking: false,
        uid: null,
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>,
    );
    expect(wrapper.find('.login-container').length).toBe(1);
  });
  test('should render the calendar screen', () => {
    store = mockStoreProvider({
      auth: {
        checking: false,
        uid: 'mock uid',
      },
      calendar: {
        events: [],
      },
      ui: {
        modalOpen: false,
      }
    });
    store.dispatch = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>,
    );
    expect(wrapper.find('.calendar-screen').length).toBe(1);
  });
});
