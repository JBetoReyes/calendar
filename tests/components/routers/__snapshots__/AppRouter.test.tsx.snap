// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter should match with snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AppRouter)>
    <AppRouter
      checking={true}
      renewToken={[Function]}
      uid={null}
    >
      <h5>
        Waiting...
      </h5>
    </AppRouter>
  </Connect(AppRouter)>
</Provider>
`;
